### DISCUSSION TITLE
Open modal using a custom function

### DISCUSSION THREAD

**Discussion by bykiandra** [mantinedev/mantine#4263] [Q&A] [answered]
So I want to display either a success or failure message in a modal after an email form has been submitted. I have the modal popping up when the submit button is clicked.

However, because the form takes time to be sent (I'm using React and Axios), even if the email is sent successfully, the error message pops up in the modal because the email hasn't finished sending yet.

I was thinking useEffect would be the best thing to use here, but how do I actually trigger the modal to open without any additional clicks or actions by the user?

I've looked through the other discussions, and while there were some that looked promising, I haven't been able to find one that works for my situation.

**Comment by rtivital** [comment]
`yourAction().then(() => success()).catch(() => fail())`

**Reply by bykiandra** [reply]
Thanks for the reply. I can get that far, but how do I actually call for the modal to be opened within my functions? Kind of like how you would call setState(), but with the modals.

**Reply by bykiandra** [reply, marked as answer]
No worries, figured it out by using it in conjunction with setState.
